src/core/textrenderer/qgstextformat.cpp

```diff
@@ -405,7 +406,7 @@ void QgsTextFormat::readFromLayer( QgsVectorLayer *layer )
 {
   d->isValid = true;
   QFont appFont = QApplication::font();
-  mTextFontFamily = layer->customProperty( QStringLiteral( "labeling/fontFamily" ), QVariant( appFont.family() ) ).toString();
+  mTextFontFamily = QgsApplication::fontManager()->processFontFamilyName( layer->customProperty( QStringLiteral( "labeling/fontFamily" ), QVariant( appFont.family() ) ).toString() );
```

Comment:
What if a project is consumed by multiple users, and one user has a unique font family replacement. Should we instead keep the original font value here and only do the replacement upon rendering? That's a question that applies whenever we read from layer, XML, etc.

Reply:
In short: I'm not sure. I can see the value both ways. Let's push forward with the current approach and see what kind of user feedback we get after this has been in the wild for a while.

Reply:
@nyalldawson , ok, sounds reasonable.